php/datasource/class-fieldmanager-datasource-user.php

```diff
@@ -44,22 +57,56 @@ class Fieldmanager_Datasource_User extends Fieldmanager_Datasource {
 
     /**
      * @var string|Null
-     * If not empty, set this post's ID as a value on the user. This is used to
+     * If not empty, set this object's ID as a value on the user. This is used to
      * establish two-way relationships.
      */
     public $reciprocal = Null;
 
-    // constructor not required for this datasource; options are just set to keys,
-    // which Fieldmanager_Datasource does.
+    /**
+	 * Constructor. Used for validation.
+	 */
+	public function __construct( $options = array() ) {
+		parent::__construct( $options );
+		
+		// Validate improper usage of store property
+		if ( ! in_array( $this->store_property, $this->allowed_store_properties ) ) {
+			throw new FM_Developer_Exception( sprintf( 
+				__( 'Store property %s is invalid. Must be one of %s.', 'fieldmanager' ),
+				$this->store_property,
+				implode( ', ', $this->allowed_store_properties )
+			) );
+		}
+		
+		if ( ! empty( $this->reciprocal ) && 'ID' != $this->store_property ) {
+			throw new FM_Developer_Exception( __( 'You cannot use reciprocal relationships with FM_Datasource_User if store_property is not set to ID.', 'fieldmanager' ) );
+		}
+	}
 
     /**
-     * Get a post title by post ID
+     * Get a user by the specified field.
      * @param int $value post_id
-     * @return string post title
+     * @return WP_User
      */
     public function get_value( $value ) {
-        $id = intval( $value );
-        $user = get_userdata( $id );
+    	switch ( $this->store_property ) {
+    		case 'ID':
+    			$field = 'ID';
+    			break;
+    		case 'user_nicename':
+    			$field = 'slug';
+    			break;
+    		case 'user_email':
+    			$field = 'email';
+    			break;
+    		case 'user_login':
+    			$field = 'login';
+    			break;
+    	}
+    	
+    	// Sanitize the value
+    	$value = $this->sanitize_value( $value );
+        
+        $user = get_user_by( $field, $value );
         return $user ? $user->{$this->display_property} : '';
```

Comment:
Looks like some of the above lines might be space-indented -- they don't look quite right